drracket/help/private/bug-report-controls.rkt

```diff
@@ -368,34 +339,28 @@
   
   
   (define (sanity-checking)
-    (let ([no-value?
-           (lambda (f)
-             (cond
-               [(is-a? f editor-canvas%)
-                (= 0 (send (send f get-editor) last-position))]
-               [else (string=? "" (send f get-value))]))])
-      (let/ec done-checking
-        (for-each
-         (lambda (field field-name)
-           (when (no-value? field)
-             (message-box (string-constant illegal-bug-report)
-                          (format (string-constant pls-fill-in-field) field-name))
-             (done-checking #f)))
-         (list name summary)
-         (list (string-constant bug-report-field-name)
-               (string-constant bug-report-field-summary)))
-        
-        (when (and (no-value? description)
-                   (no-value? reproduce))
-          (message-box (string-constant illegal-bug-report)
-                       (string-constant pls-fill-in-either-description-or-reproduce))
-          (done-checking #f))
-        
-        (unless (regexp-match #rx"@" (or (preferences:get 'drracket:email) ""))
-          (message-box (string-constant illegal-bug-report)
-                       (string-constant malformed-email-address))
-          (done-checking #f))
-        (done-checking #t))))
+    (define (no-value? f)
+      (cond
+        [(is-a? f editor-canvas%) (= 0 (send (send f get-editor) last-position))]
+        [else (string=? "" (send f get-value))]))
+    (let/ec done-checking
+            (for-each (lambda (field field-name)
+                        (when (no-value? field)
+                          (message-box (string-constant illegal-bug-report)
+                                       (format (string-constant pls-fill-in-field) field-name))
+                          (done-checking #f)))
+                      (list name summary)
```

Comment:
Personally I would write it like this:

```
(define (<f> field field-name)
  ...)

(<f> name (string-constant bug-report-field-name))
(<f> summary (string-constant bug-report-field-summary))
```

It's going to be difficult to come up with the name `<f>` though...

Reply:
Agreed, but yeah there's no easy way to automatically pick a name for `<f>` here.

Unless... I suppose... one _could_ use an LLM to do that...

No. I will not walk down that dark road.

Reply:
a `for` loop still seems preferable to the `for-each`, tho?